proposals/low-level-struct-improvements.md

```diff
@@ -805,25 +730,119 @@ For example the last line of calculating *safe-to-escape* of returns will change
+
+For example, the signature of the indexer for `CharBuffer.Data` will be the following:
+
+```c#
+[RefThisEscapes]
+internal ref char <>DataIndexer(int index) => ...;
+```
+
+If the provided index is outside the declared bounds of the `fixed` array then an `IndexOutOfRangeException` will be thrown. In the case a constant value is provided then it will be replaced with a direct reference to the appropriate element. Unless the constant is outside the declared bounds in which case a compile time error would occur.
+
+There will also be a named accessor generated for each `fixed` buffer that provides by value `get` and `set` operations. Having this means that `fixed` buffers will more closely resemble existing array semantics by having a `ref` accessor as well as byval `get` and `set` operations. This means compilers will have the same flexibility when emitting code consuming `fixed` buffers as they do when consuming arrays. This should be operations like `await` over `fixed` buffers easier to emit. 
+
+This also has the added benefit that it will make `fixed` buffers easier to consume from other languages. Named indexers is a feature that has existed since the 1.0 release of .NET. Even languages which cannot directly emit a named indexer can generally consume them (C# is actually a good example of this).
+
+There will also be a by value `get` and `set` accessor generated for every 
+
+The backing storage for the buffer will be generated using the `[InlineArray]` attribute. This is a mechanism discussed in [isuse 12320](https://github.com/dotnet/runtime/issues/12320) which allows specifically for the case of efficiently declaring sequence of fields of the same type. This particular issue is still under active discussion and the expectation is that the implementation of this feature will follow however that discussion goes.
+
 ## Considerations
 
 ### Keywords vs. attributes
-This design calls for using attributes to annotate the new lifetime rules for `struct` members. This also could've been done just as easily with contextual keywords. For instance: `scoped` and `escapes` could have been used instead of `DoesNotEscape` and `ThisRefEscapes`.
+This design calls for using attributes to annotate the new lifetime rules. This also could've been done just as easily with contextual keywords. For instance: `scoped` and `escapes` could have been used instead of `DoesNotEscape` and `RefThisEscapes`.
 
 Keywords, even the contextual ones, have a much heavier weight in the language than attributes. The use cases these features solve, while very valuable, impact a small number of developers. Consider that only a fraction of high end developers are defining `ref struct` instances and then consider that only a fraction of those developers will be using these new lifetime features.  That doesn't seem to justify adding a new contextual keyword to the language.
```

Comment:
Attributes though do seem to escape the language. In this case, will these attributes be meaningful in other .NET languages like VB.NET or F#? If they are applied will they be nops or having some compiler indicate they are unnecessary?

Reply:
By default these attributes won't be meaningful to other languages. But that's true today. Consider that if they put the `[Dynamic]` attribute on method parameters it has no meaning in F# but will when it's invoked from C#. 

There is a growing list of APIs for which they seem to only have meaning in C#. I'm not sure how or if we should dela with that.

Reply:
> I'm not sure how or if we should dela with that.

Okay. I'll think this does come with some concerns though. From the language it may not matter, but how the runtime may or may not respond to that attribute in the future seems to me to be a potential concern. In fact I could see a strong argument for having tests in those languages to ensure the runtime doesn't do things the generated IL doesn't expect. I don't have a concrete example but I could imagine method level attributes that let the runtime inform the JIT about some lifetime assumption that isn't there in a non-C# language and that ends up causing issues.

I'm in the middle of a GC Stress bug and it is consuming a fair amount of mental resources so it could be simple top of mind and I see potential issues everywhere.